frontend/src/components/UserProfilePanels/UserInfo.js

```diff
@@ -1,70 +1,140 @@
-        <div className="col-md-2" align="center">
-          <img
-            src={profilePic}
-            alt="User"
-            className="profilePage-profilePic"
+function UserInfo({ user, setUser }) {
+  const [editBox, hideEditBox] = useState(false);
+  const [userFirstName, setUserFirstName] = useState(user.first_name);
+  const [userLastName, setUserLastName] = useState(user.last_name);
+  const [error, setError] = useState('');
+
+  const validateUserName = (userInputName, userInputLast) => {
+    if (userInputName === '' || userInputLast === '') {
+      setError('Please enter your first and last name.');
+      return false;
+    }
+    return true;
+  };
+
+  const editUser = async () => {
+    if (validateUserName(userFirstName, userLastName)) {
+      await editCurrentUserName(userFirstName, userLastName);
+      const newData = { first_name: userFirstName, last_name: userLastName };
+      const newUserData = { ...user, ...newData };
+      setUser(newUserData);
+      hideEditBox(false);
+    }
+  };
+
+  const renderEditBox = (isEditing) => {
+    if (isEditing === true) {
+      return (
+        <div className="edit-name">
+          <input
+            type="text"
+            defaultValue={userFirstName}
+            placeholder="First Name"
+            onChange={e => setUserFirstName(capitalize(trim(e.target.value)))}
           />
-          <div>
-            <button type="button" className="user-edit-button">
-              <img
-                src="https://img.icons8.com/windows/32/000000/edit.png"
-                alt="edit"
-              />
-            </button>
-            <button type="button" className="user-edit-button">
-              <img
-                src="https://img.icons8.com/windows/32/000000/cancel.png"
-                alt="delete"
-              />
-            </button>
-          </div>
+          <input
+            type="text"
+            defaultValue={userLastName}
+            placeholder="Last Name"
+            onChange={e => setUserLastName(capitalize(trim(e.target.value)))}
+          />
+          <button
+            type="button"
+            className="user-edit-button"
+            onClick={editUser}
+          >
+            <img
+              src={checkmark}
+              alt="save"
+              height="60"
+            />
+          </button>
+          <p>{error}</p>
         </div>
-        <div className="col-md-9 user-info">
-          <div className="profilePage-user-name">
-            <h1>
-              {user.first_name}
-              {' '}
-              {user.last_name}
-            </h1>
-            <button type="button" className="user-edit-button">
-              <img
-                src="https://img.icons8.com/windows/32/000000/edit.png"
-                alt="edit"
-              />
-            </button>
+      );
+    }
+    return (
+      <div>
+        <h1>
+          {userFirstName}
+          {' '}
+          {userLastName}
```

Comment:
NitPick: I don't think this needs to be three separate lines. It'd be a little cleaner to do something like this:
```
 { `${userFirstName} ${userLastName}` }
```